test/e2e/operator_test.go

```diff
@@ -108,71 +126,199 @@ var _ = Describe("Operator", func() {
+					err := client.Delete(clientCtx, sa)
+					if apierrors.IsNotFound(err) {
+						return nil
+					}
+					return err
+				}).Should(Succeed())
+			}(sa)
 		}
 
 		// Label sa-a and sa-b with o's component label
-		saA.SetLabels(map[string]string{expectedKey: ""})
-		_, err = kubeClient.CoreV1().ServiceAccounts(saA.GetNamespace()).Update(clientCtx, saA, updateOpts)
-		Expect(err).ToNot(HaveOccurred())
-		saB.SetLabels(map[string]string{expectedKey: ""})
-		_, err = kubeClient.CoreV1().ServiceAccounts(saB.GetNamespace()).Update(clientCtx, saB, updateOpts)
-		Expect(err).ToNot(HaveOccurred())
+		Eventually(Apply(saA, setComponentLabel)).Should(Succeed())
+		Eventually(Apply(saB, setComponentLabel)).Should(Succeed())
 
 		// Ensure o's status.components.refs field eventually contains references to sa-a and sa-b
 		By("eventually listing multiple component references")
-		componentRefEventuallyExists(w, true, saA.GetName())
-		componentRefEventuallyExists(w, true, saB.GetName())
+		componentRefEventuallyExists(w, true, getReference(scheme, saA))
+		componentRefEventuallyExists(w, true, getReference(scheme, saB))
 
 		// Remove the component label from sa-b
-		saB.SetLabels(nil)
-		_, err = kubeClient.CoreV1().ServiceAccounts(saB.GetNamespace()).Update(clientCtx, saB, updateOpts)
-		Expect(err).ToNot(HaveOccurred())
+		Eventually(Apply(saB, func(m metav1.Object) error {
+			m.SetLabels(nil)
+			return nil
+		})).Should(Succeed())
 
 		// Ensure the reference to sa-b is eventually removed from o's status.components.refs field
 		By("removing a component's reference when it no longer bears the component label")
-		componentRefEventuallyExists(w, false, saB.GetName())
+		componentRefEventuallyExists(w, false, getReference(scheme, saB))
 
 		// Delete ns-a
-		Expect(kubeClient.CoreV1().Namespaces().Delete(clientCtx, nsA.GetName(), deleteOpts)).To(Succeed())
+		Eventually(func() error {
+			err := client.Delete(clientCtx, nsA)
+			if apierrors.IsNotFound(err) {
+				return nil
+			}
+			return err
+		}).Should(Succeed())
 
 		// Ensure the reference to ns-a is eventually removed from o's status.components.refs field
 		By("removing a component's reference when it no longer exists")
-		componentRefEventuallyExists(w, false, nsA.GetName())
+		componentRefEventuallyExists(w, false, getReference(scheme, nsA))
+	})
+
+	Context("when a subscription to a package exists", func() {
+		var (
+			ns           *corev1.Namespace
+			sub          *operatorsv1alpha1.Subscription
+			operatorName types.NamespacedName
+		)
+
+		BeforeEach(func() {
+			// Subscribe to a package and await a successful install
+			ns = &corev1.Namespace{}
+			ns.SetName(genName("ns-"))
+			Eventually(func() error {
+				return client.Create(clientCtx, ns)
+			}).Should(Succeed())
+
+			// Default to AllNamespaces
+			og := &operatorsv1.OperatorGroup{}
+			og.SetNamespace(ns.GetName())
+			og.SetName(genName("og-"))
+			Eventually(func() error {
+				return client.Create(clientCtx, og)
```

Comment:
this line is failing in e2e tests

```
    Expected success, but got an error:
        <*errors.StatusError | 0xc000b07e00>: {
            ErrStatus: {
                TypeMeta: {Kind: "Status", APIVersion: "v1"},
                ListMeta: {
                    SelfLink: "",
                    ResourceVersion: "",
                    Continue: "",
                    RemainingItemCount: nil,
                },
                Status: "Failure",
                Message: "OperatorGroup in version \"v1\" cannot be handled as a OperatorGroup: unmarshalerDecoder: Object 'Kind' is missing in '{\"metadata\":{\"name\":\"og-mjsp5\",\"namespace\":\"ns-cpnst\",\"creationTimestamp\":null},\"spec\":{},\"status\":{\"lastUpdated\":null}}', error found in #10 byte of ...|ed\":null}}\n|..., bigger context ...|mp\":null},\"spec\":{},\"status\":{\"lastUpdated\":null}}\n|...",
                Reason: "BadRequest",
                Details: nil,
                Code: 400,
            },
        }
        OperatorGroup in version "v1" cannot be handled as a OperatorGroup: unmarshalerDecoder: Object 'Kind' is missing in '{"metadata":{"name":"og-mjsp5","namespace":"ns-cpnst","creationTimestamp":null},"spec":{},"status":{"lastUpdated":null}}', error found in #10 byte of ...|ed":null}}
        |..., bigger context ...|mp":null},"spec":{},"status":{"lastUpdated":null}}
        |...
```

I'm guessing the client doesn't have the v1 api in it's scheme - or you can just add typemeta here in the test

Reply:
Turns out the helm provisioner was setting some fields on the rest config that were causing the controller-runtime client to use a different -- failing -- codec